### PR TITLE
NEUBURG-77 use new error format

### PR DESCRIPTION
**Pull request by michael-markl**
This pull request belongs to an issue on our [bugtracker](https://integreat.atlassian.net/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/routes/living-manage-offer/components/ConfirmOfferManager.js [hunk 1]
```diff
@@ -2,20 +2,34 @@ import React from 'react'
...
     // Immediately send confirmRequest
     this.props.send('POST', `/confirm`)
   }
 
+  getErrorMessage () {
+    switch (this.props.serverError.status) {
+      case GONE:
+        return 'Der Bestätigungslink ist ungütltig.'
+      case NOT_FOUND:
+        return 'Konnte zugehöriges Angebot nicht finden.'
```
Comment (steffenkleinle) [inline src/routes/living-manage-offer/components/ConfirmOfferManager.js]: Change to: "Angebot konnte nicht gefunden werden." or something like that.

File: src/routes/living-manage-offer/components/DeleteOfferManager.js [hunk 2]
```diff
@@ -15,6 +16,14 @@ export class DeleteOfferManager extends React.Component {
     this.props.send('DELETE')
   }
 
+  getErrorMessage () {
+    switch (this.props.serverError.status) {
+      case NOT_FOUND:
+        return 'Konnte zugehöriges Angebot nicht finden.'
```
Comment (steffenkleinle) [inline src/routes/living-manage-offer/components/DeleteOfferManager.js]: Same as above.

File: src/routes/living-manage-offer/components/ExtendOfferManager.js [hunk 3]
```diff
@@ -18,9 +20,20 @@ export class ExtendOfferManager extends React.Component {
...
+      return {duration: values.duration}
+    }, () => {
+      this.props.send('POST', '/extend', this.state.form.getValues())
+    })
+  }
+
+  getErrorMessage () {
+    switch (this.props.serverError.status) {
+      case NOT_FOUND:
+        return 'Konnte zugehöriges Angebot nicht finden.'
```
Comment (steffenkleinle) [inline src/routes/living-manage-offer/components/ExtendOfferManager.js]: Same, maybe we could even extract this method and reuse it?

File: src/routes/living-manage-offer/containers/LivingManageOfferPage.js [hunk 4]
```diff
@@ -30,14 +29,25 @@ export class LivingManageOfferPage extends React.Component {
...
+            }
+          }))
         }
       })
       .catch(() => {
-        this.setState({success: false, sending: false, serverError: 'Verbindung fehlgeschlagen'})
+        this.setState({
+          success: false,
+          sending: false,
+          serverError: {status: 0, message: 'Die Verbindung ist fehlgeschlagen.'}
```
Comment (steffenkleinle) [inline src/routes/living-manage-offer/containers/LivingManageOfferPage.js:49]: Hardcoded message, but stuff like this is not important here right?

**Review by steffenkleinle — APPROVED** [review]

File: src/routes/living-manage-offer/containers/LivingManageOfferPage.js (hunk 4 above)
Comment (michael-markl) [reply src/routes/living-manage-offer/containers/LivingManageOfferPage.js:49]: Hmm.. actually it is important... I haven't payed attention to that yet -.- I'll create a task to refactor hardcoded text 👍

**Review by maxammann — APPROVED** [review]

File: src/routes/living-manage-offer/components/ConfirmOfferManager.js [hunk 5]
```diff
@@ -2,20 +2,34 @@ import React from 'react'
...
 
   componentWillMount () {
     // Immediately send confirmRequest
     this.props.send('POST', `/confirm`)
   }
 
+  getErrorMessage () {
+    switch (this.props.serverError.status) {
+      case GONE:
+        return 'Der Bestätigungslink ist ungütltig.'
```
Comment (NKlug) [inline src/routes/living-manage-offer/components/ConfirmOfferManager.js:26]: Why don't you use the servers error messages here? I mean we already have them :)

**Review by steffenkleinle — APPROVED** [review]

**Pull request merged**